### PR TITLE
dbscripts_on_relay and randomization

### PR DESCRIPTION
**Pull request by killerwife**
This PR introduces dbscripts_on_relay which enable launching scripts from EAI, C++ or other dbscripts for example.

This also enables random scripting with waypoints. Minor refactor to use one joined structure for string and relay randomization.

Last but not least, rename of db_script_string into dbscript_string to keep a unified naming logic.

All of the newly added stuff is possible from dbscripts, EAI and SD2 alike.

@grz3s @xfurry @cala

It has been tested by @mantislord and @saltgurka already.

### CODE REVIEW

File: src/game/AI/EventAI/CreatureEventAIMgr.cpp [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "Maps/GridDefines.h"
 #include "AI/ScriptDevAI/ScriptDevAIMgr.h"
 #include "World/World.h"
+#include "DBScripts\ScriptMgr.h"
```
Comment (Inifield) [inline src/game/AI/EventAI/CreatureEventAIMgr.cpp]: this bothers me more than it should but could you change "\" to "/"

Comment (killerwife) [inline src/game/AI/EventAI/CreatureEventAIMgr.cpp]: I will, thank you for the remark.

**Comment by killerwife** [comment]
Added two more events, refined documentation, will merge this next week if no opposition is met.

File: sql/updates/mangos/06Dbscript_random_template.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+) ENGINE=InnoDB DEFAULT CHARSET=utf8 ROW_FORMAT=COMPACT COMMENT='DBScript system';
+
+-- type 0 - strings
+-- type 1 - relay dbscript
+
+INSERT INTO dbscript_random_templates(id,type,target_id,chance) SELECT id,'0',string_id,'0' FROM dbscript_string_template;
+
+DROP TABLE IF EXISTS dbscript_string_template;
+
+DROP TABLE `dbscript_string`;
```
Comment (Saltgurka) [inline sql/updates/mangos/06Dbscript_random_template.sql]: DROP TABLE dbscript_string; -> DROP TABLE __IF EXISTS__ dbscript_string;

**Comment by killerwife** [comment]
Merged.

**Pull request closed without merging**